### DISCUSSION TITLE
Create Additions (Rotation -> FE, FE -> Rotation)

### DISCUSSION THREAD

**Discussion by merlerm** [NillerMedDild/Enigmatica6#1190] [Ideas]
This has been discussed countless times but it seems to pop up again and again so I thought I'd open up a discussion here to settle this once and for all and make up our minds about it.

https://www.curseforge.com/minecraft/mc-mods/createaddition

Create Additions would allow to convert rotation into FE and viceversa. While power -> rotation might not be in the spirit of the pack, it's at least worth mentioning because I think it could be helpful for people who don't really want to play with create, as it allows them a way to "cheat" the system, but it could very easily be disabled.

The big one is rotation to FE here. It would allow really cool setups and let us createheads make a base running only on rotation, which imo is a really fun concept and something I could see a lot of people trying. With expert in mind, it might also be a cool way to gate every sort of powergen, making people do at least a bit of create before they can power their first electric machine and then upgrade from that.
The concerns with this are infinite power loops, already found by Motoko, due to interactions with other mods. I'm not sure about the values myself so it's something we should look into and see how big of a problem it is. Personally as long as it isn't too crazy I'd be fine with it: it would require clever setups anyways, and passive/infinite powergen is already a thing in the pack with the various windmills/watermills/solars etc. If we can somehow get the numbers in line with those (the mod seems fairly configurable), we should have no issues there.

It also adds a plate/rod maker that works with create rotation, which is interesting in the scope of expert mode again imo, since it could be the earliest way to automate them, similar to the metal press in e2e. Something to keep in mind

It's also worth noting that Create Integrations (https://www.curseforge.com/minecraft/mc-mods/create-integration) exists, that fills much of the same purpose but with some extra neat gadgets, like chunkloaders and ender chests that can be stuck to a moving contraption. It's only 1.15 so far so I'm not sure how relevant including it here is, but it might be worth keeping an eye on in the future for those extra gadgets. 

What do you guys think?

**Comment by KrAzYGEEK32** [comment]
so.... here is my take on this. i personally like how PNC handles FE conversions by gating them behind some top tier stuff like the assembly line. i dont see how FE-rotation wouldnt be in the right spirit of the pack if you think about it. you are just replacing the source of rotation not the mechanics of Create (generating rotational force is like 5% of the mod imo). but as motoko mentioned multiple times before, rotation to FE could be abused a lot for infinite power, not that it would be effective but still it doesnt make sense that thats a thing. for example, you make su(rotational power) via an encased fan, that could probably run an IE external heater, now that removes the upkeep cost to run a furnace engine which makes 16ksu, so you basically generate 16ksu from 16su. thats... just ridiculous imo :). so i suggest keep the FE -> SU but gate it behind brass and mechanical crafting but dont add SU->FE because that would be broken with the external heater being a thing(unless you could somehow disable that functionality of the external heater).

TLDR: keep FE->SU, and remove SU->FE because of it being broken due to intermod interaction.

**Reply by merlerm** [reply]
I don't agree with this - rotation to FE is the main appeal here for me and I think we should try to keep it. The other way around is cool, but SU -> FE is unique and there's really not another way to do the same. I would need to see how broken it was first because if the FE generation you got from the loop was only a couple hundreds FE/t I don't really see an issue with it. As for the infinite rotation loop, that's why I'd have said disable FE -> SU instead.

**Reply by KrAzYGEEK32** [reply]
I don't understand how disabling fe->su would solve the infinite power loop..

**Comment by NexusNull** [comment]
Honestly, even if there are infinity power loops as long as they aren't that overpowered it's fine, we have solar panels after all which is an infinite power loop in a box.

**Reply by KrAzYGEEK32** [reply]
well, atleast solar panels have some semblence of logic to them and they dont run at night :P

**Reply by MuteTiefling** [reply]
Solar also provides a fixed amount. You don't suddenly get a 10x output increase just by sticking two of them next to each other.

**Comment by MuteTiefling** [comment]
I think people are underestimating the power loop here. 

IE's External Heater currently requires about 7rf/t to run a single Create Furnace engine. 

A single Thermoelectric Generator from IE, for instance, should do about 42 rf/t with packed ice and uranium. That will run 6 furnace engines. At base configs, the furnace engine will run 4 alternators for a total of 180 rf/t. so for 42 rf/t in, you get 1080rf/t out. 

This gets worse when you use a Blast Furnace and PNC to heat it (IE won't) as the Blast Furnace generates twice as much SU, so 2160 rf/t for about 0.2 bar of air... PNC also has a compressor that works on temperature differentials. A torch and a heat sink would probably be enough to supply that tiny amount of air. 


Now, I think we can all agree that 180 rf/t doesn't get you very far with most machines in the pack. I don't think that would even consistently run most thermal machines with any upgrades. Maybe some mek stuff with lots of energy upgrades and no speed upgrades. So obviously this would have to go up to even be of much use without powering hundreds of alternators, making the situation worse. 


Disabling this power loop means disabling core features of Pneumaticcraft (heat powering furnaces) and disabling the external heater from IE (or nerfing it into uselessness, as it would be if it actually required more than 1080 rf/t to run).

The only thing of real interest to me is the Rolling Mill, honestly. As that would allow getting into Immersive without the Immersive Plate Press.

**Reply by merlerm** [reply]
I'm surprised the external heater is so cheap, even if the mod doesn't get added, wouldn't it be fair to make it cost a bit more power for what's essentially free smelting? I'm not sure what the other smelting alternatives cost but I'm sure TE's redstone furnace doesn't take 7 fe/t for example.

If the loop is that bad I guess I'll have to call quits on this but it's still super appealing to me to be able to generate power off of SU. Even nerfing the generator's rates wouldn't work well, because then it'd become unusable.

I'm pretty sure I already know the answer is no, but could there be any way to blacklist the flywheel specifically from the heaters? Even if there's a blacklist feature at all, is it considered a different block/entity?

**Reply by MuteTiefling** [reply]
The alternator couldn't possibly know what's powering it, I think. And the heater isn't interacting with the engine directly, but the vanilla furnace that's powering the engine

**Reply by merlerm** [reply]
Yeah, I was wondering if that furnace had some special value on it we could use to figure out it was fueling the engine and not a normal furnace in particular

**Reply by OneLemonyBoi** [reply]
> Yeah, I was wondering if that furnace had some special value on it we could use to figure out it was fueling the engine and not a normal furnace in particular

Just checked using /data, and there is no special value.

**Reply by AbsorbingSafe11** [reply]
(Sorry if message was too late or already known)
The Thermoelectric Generator from IE uses up the blocks placed next to it, like packed ice into water eventually and uranium into lead. (also I thought uranium was an expensive material and you would require blocks of it for that setup)

**Comment by willbattel** [comment]
Is it possible to configure the conversion efficiency? I understand the infinite power concern, but if it can be configured such that the net gain isn't totally ridiculous like it seems to be by default then I don't think it'd be a problem. Infinite power loops have been around in tech-heavy modpacks forever and are even romanticized in cases where a complex contraption is required to do it.

I agree that a simple loop that spits out stupid infinite energy should be avoided, but if the efficiency and recipes can be nerfed sufficiently then this mod would be great for people that want to put their stockpiled FE to work in Create. After all, energy is energy. Conversions are lossy but if I have a large source of FE then I don't think I should have to make a water/windmill to use Create, and vice versa.

I also empathize with the argument that bypassing energy production in one mod by converting from another may not be in the spirit of the mods and/or this pack. Therefore, in my eyes, either choice makes sense and it all comes down to personal preference. My personal preference is to add this conversion mod and nerf the efficiency if possible as well as make the recipes appropriately difficult so that you can't skip in the early-game.

**Comment by ELEKTRIGGER** [comment]
Hope this not too necro...but its a simple exploitation  to consume such less fe for Heat Produktion, keep in Mind that smeltering / Metsl Industrie  is consuming MegaWatts ....
